tasks/modules/compile.js

```diff
@@ -63,33 +64,34 @@ function getTsc(binPath) {
     exports.grunt.log.writeln('Using tsc v' + pkg.version);
     return path.join(binPath, 'tsc');
 }
-function compileAllFiles(targetFiles, target, task, targetName, outFile) {
+function compileAllFiles(options, compilationInfo) {
+    var targetFiles = compilationInfo.src;
     // Make a local copy so we can modify files without having external side effects
     var files = _.map(targetFiles, function (file) { return file; });
     var newFiles = files;
-    if (task.fast === 'watch') {
+    if (options.fast === 'watch') {
         // if this is the first time its running after this file was loaded
         if (cacheClearedOnce[exports.grunt.task.current.target] === undefined) {
             // Then clear the cache for this target
-            clearCache(targetName);
+            clearCache(options.targetName);
         }
     }
-    if (task.fast !== 'never') {
-        if (target.out) {
+    if (options.fast !== 'never') {
+        if (compilationInfo.out) {
             exports.grunt.log.writeln('Fast compile will not work when --out is specified. Ignoring fast compilation'.cyan);
         }
         else {
-            newFiles = getChangedFiles(files, targetName);
-            if (newFiles.length !== 0) {
+            newFiles = getChangedFiles(files, options.targetName);
+            if (newFiles.length !== 0 || options.testExecute) {
                 files = newFiles;
                 // If outDir is specified but no baseDir is specified we need to determine one
-                if (target.outDir && !target.baseDir) {
-                    target.baseDir = utils.findCommonPath(files, '/');
+                if (compilationInfo.outDir && !options.baseDir) {
+                    options.baseDir = utils.findCommonPath(files, '/');
                 }
             }
             else {
                 exports.grunt.log.writeln('No file changes were detected. Skipping Compile'.green);
```

Comment:
Same: this error will trigger in the tsconfig passThrough case. Perhaps another `||` clause for that case?